Homework/Program.cs

```diff
@@ -1,10 +1,84 @@
-﻿namespace Homework
+﻿using System.Text.Json;
+
+namespace Homework 
 {
-    internal class Program
+    public class Program
     {
-        static void Main(string[] args)
+        public static async Task Main(string[] args)
         {
-            Console.WriteLine("Hello, World!");
+            Console.WriteLine();
+            Console.WriteLine(" PROGRAM NA ZISTENIE VÝCHODU-ZÁPADU SLNKA a MESIACA");
+            Console.WriteLine();
+            string jsonString = await getPublicApiDataAsync();
+            if (jsonString != null) deserializeJsonMessage(jsonString); else Console.WriteLine(" Odpoveď je prázdna - špatné parametre na vstupe ");
+        }
+
+        // Ukazka volani https://www.weatherapi.com/api-explorer.aspx#current API
+        // pro vyzkouseni musime mit vlastni validni apiKey. Ten ziskate po bezplatne registraci
+        public static async Task<string> getPublicApiDataAsync()
+        {
+            string responseBody = null;
+            try
+            {
+                using HttpClient client = new HttpClient();
+
+                // url adresa api vcetne parametru
+                const string apiKey = "d0f1dd626e1c4705af0164004250401";
+                string city = null;
+                Console.Write("Zadaj mesto (prázdne=Senica): ");
+                city = Console.ReadLine();
+                if (city == "") { city = "Senica"; }
+                DateTime dnesnyDatum = DateTime.Now;
+                int dnesnyRok = dnesnyDatum.Year;
+                int dnesnyMes = dnesnyDatum.Month;
+                int dnesnyDay = dnesnyDatum.Day;
+                string dnesnyDatumInak = $"{dnesnyRok}-{dnesnyMes}-{dnesnyDay}";
+                Console.Write($"Zadaj dátum v tvare (2025-01-15)(prázdne={dnesnyDatumInak}): ");
```

Comment:
Jen pro zajimavost - tento zapis (tvar) se zapisuje takto
```suggestion
                Console.Write($"Zadaj dátum v tvare YYYY-MM-DD(prázdne={dnesnyDatumInak}): ");
```